gamemode/modules/animations/sh_animations.lua

```diff
@@ -63,7 +63,7 @@ hook.Add("CalcMainActivity", "darkrp_animations", function(ply, velocity) -- Usi
 
     -- Hobo throwing poop!
     local Weapon = ply:GetActiveWeapon()
-    if RPExtraTeams[ply:Team()] and RPExtraTeams[ply:Team()].hobo and not ply.ThrewPoop and IsValid(Weapon) and Weapon:GetClass() == "weapon_bugbait" and ply:KeyDown(IN_ATTACK) then
+    if RPExtraTeams[ply:Team()] and RPExtraTeams[ply:Team()].hobo and not ply.ThrewPoop and Weapon:IsValid() and Weapon:GetClass() == "weapon_bugbait" and ply:KeyDown(IN_ATTACK) then
```

Comment:
Are you 100% sure `ply:GetActiveWeapon()` won't return `nil` when the player has no weapons? Shit like that is why I use `IsValid`. I hate micro-optimisations like this because they don't improve performance and are a potential for much bigger problems: error messages about shit being nil.

Reply:
I can do a quick test.

Reply:
] lua_run print(Entity(1):GetActiveWeapon())
> print(Entity(1):GetActiveWeapon())...
[NULL Entity]

Doesn't seem to.